sdk/include/opentelemetry/sdk/trace/tracer_provider.h

```diff
@@ -35,15 +49,30 @@ class TracerProvider final : public opentelemetry::trace::TracerProvider
    */
   void SetProcessor(std::shared_ptr<SpanProcessor> processor) noexcept;
 
+  /**
+   * Set the sampler associated with this tracer provider.
+   * Also update the sampler pointer stored in the corrisponding tracer.
+   * @param sampler The new sampler for this tracer provider. This
+   * must not be a nullptr.
+   */
+  void SetSampler(std::shared_ptr<Sampler> sampler) noexcept;
```

Comment:
I think we can omit all `SetSampler` methods and just go with the ability to set the sampler on `Tracer` and `TracerProvider` initialization. Other SDKs (I checked Go, Java and Python) don't have the ability to change the sampler on the fly, so we should be good without that too. At least I'm not aware of any requirement that would demand that.